ocaml/xapi/xapi_mgmt_iface.ml

```diff
@@ -28,6 +28,30 @@ let update_mh_info interface =
   in
   ()
 
+module StunnelCb : sig
+  val add : (unit -> unit) -> unit
+
+  val execute : unit -> unit
+end = struct
+  let m = Mutex.create ()
+
+  let stunnel_cb = ref None
```

Comment:
There may be a case where, by mistake (a programming error, mainly), after resetting the reference to None, the callback is set again. This would defeat the purpose of the module.

Can we add a reference to a boolean in here that 'taints' the module? Meaning once it's set to `true` it's never set to `false`.

Something like `let used = ref false`, and then
```
let used = ref false

let add cb = 
  Mutex.execute m @@ fun () ->
  if not !used && Option.is_none !stunnel_cb then (
     used := true ;
     stunnel_cb := Some cb
  )


let execute () = 
  Mutex.execute m @@ fun () ->
  match !used, !stunnel_cb with
  | true, _ | _, None -> ()
  | false, Some cb -> ...
```

This could be done by just changing the `~finally` to `fun () -> stunnel_cb := Some (Fun.const ());`, but I feel that would make it  harder to understand the intent of the module.